src/server/views/dashboard/macros.njk

```diff
@@ -98,9 +98,9 @@
         {% endif %}
         {% if "update" in actionButtons %}
         <div class="govuk-radios__item">
-          <input class="govuk-radios__input" id="action-update" name="action" type="radio" value="update" form="mainForm">
+          <input class="govuk-radios__input" id="action-update" name="action" type="radio" value="{% if listItem.isPublished %}update{% else %}publish{% endif %}" form="mainForm">
```

Comment:
`{{ 'update' if listItem.isPublished else 'publish' }}` is more readable. I would leave if / else blocks when you're dealing with multiple lines